app/javascript/packages/memorable-date/index.spec.ts

```diff
@@ -160,6 +166,75 @@ describe('MemorableDateElement', () => {
       expect(formElement.reportValidity()).to.be.true();
     });
   }
+
+  // This is for a New Relic bug that overrides
+  // the addEventListener and removeEventListener functions.
+  // See here: https://discuss.newrelic.com/t/javascrypt-snippet-breaks-site/52188
+  function itIsUnaffectedByNewRelicEventBug() {
+    context(
+      'another script overrides the addEventListener in a way that loses function identity',
+      () => {
+        let originalAddEventListenerFunction;
+        beforeEach(() => {
+          originalAddEventListenerFunction = Element.prototype.addEventListener;
+          Element.prototype.addEventListener = function addEventListener(type, listener, ...args) {
+            if (listener instanceof Function) {
+              listener = function overrideListener(...eventArgs) {
+                return listener.apply(this, eventArgs);
+              };
+            }
+
+            if (arguments.length > 1) {
+              args.unshift(listener);
+            }
+
+            if (arguments.length > 0) {
+              args.unshift(type);
+            }
+            return originalAddEventListenerFunction.apply(this, args);
+          };
+        });
+
+        afterEach(() => {
+          Element.prototype.addEventListener = originalAddEventListenerFunction;
+          originalAddEventListenerFunction = null;
+        });
+
+        context(
+          'user has entered a day and year, then moves clicks an element outside the memorable date fields',
```

Comment:
```suggestion
          'user has entered a day and year, then clicks an element outside the memorable date fields',
```

Reply:
Is there a reason this set of tests only cares about the day and year, and not the month?

Reply:
The bug is that `removeEventListener` isn't actually removing the event listener because New Relic changes the function identity passed into the original `removeEventListener`. This is a workaround for how that bug manifests in the new date field.

Reply:
But can you explain how to reproduce the bug so that I can test it?

Reply:
Here's a code snippet that reproduces the issue when I tested on another input:

```
const clickListener = function() {
    console.log('clicked');
    this.removeEventListener('click', clickListener);
    console.log('dispatching');
    this.dispatchEvent(new CustomEvent('click', { bubbles: true}));
    console.log('dispatched');
    this.addEventListener('click', clickListener);
    console.log('done');
};
const blurListener = () => console.log('clicked 2');

// $0 is the selected <input> element
$0.addEventListener('click', clickListener)
$0.addEventListener('blur', blurListener)
```

Reply:
Maybe it's not the function identity exactly, but this does sound like something in the New Relic changelog.

Reply:
Still haven't been able to reproduce the bug myself, would love to do that so that I feel confident that I fully understand the changes and the original bug. Have been DMing with Tim about steps to reproduce. Doesn't have to block the PR though

Reply:
Tim and I paired and talked through this -- thank you. The bug is somewhat intermittent